### ISSUE TITLE
False alarm for chained-comparison

### ISSUE THREAD

**Issue by kayhayen** [PyCQA/pylint#2305]
### Steps to reproduce
1.
```
def isAtLeastSubVersion(version):
    if version < 280 and \
       280 <= python_version < 300:
        return True

    if version // 10 != python_version // 10:
        return False

    return python_version >= version

```

### Current behavior

> chained-comparison isAtLeastSubVersion Simplify chained comparison between the operands

pointing to the first statement of the function

### Expected behavior

No such simplification is possible, since "version" and "python_version" are distinct variables.

### pylint --version output

> 
> python -m pylint --version
> __main__.py 2.0.0
> astroid 2.0.0.dev4
> Python 3.6.5 (v3.6.5:f59c0932b4, Mar 28 2018, 16:07:46) [MSC v.1900 32 bit (Intel)]

Python 3.6.5 (v3.6.5:f59c0932b4, Mar 28 2018, 16:07:46) [MSC v.1900 32 bit (Intel)]

**Comment by sushobhit27** [comment]
@kayhayen because above the expression can be further chained like:

```python
def isAtLeastSubVersion(version):
    if version < 280 <= python_version < 300:
        return True

    if version // 10 != python_version // 10:
        return False

    return python_version >= version
```

copy pasting original code in pycharm also highlights same issue although it does not identify how to chain it.

![image](https://user-images.githubusercontent.com/31987769/42774943-bd446642-8950-11e8-9065-0908083ef86a.png)

**Comment by kayhayen** [comment]
Woah, that's gross. I wouldn't want that, so the 280 is common, and that is enough?

"a > 1 and b > 1" should really be written as  "a < 1 > b" seriously?

But I admit, this bug report is bogus. I am closing it.

**Comment by sushobhit27** [comment]
@kayhayen yeah I know a < 1 < b looks weird due to variable not being in middle, but take an example of comparing date. In my project's codebase, many times we have to compare whether something (e.g ticker of a stock) is expired or not.

``` python
class Something:
    def get_effective_date(self):
        pass
    
    def get_expiration_date(self):
        pass
    
obj = Something()
if obj.get_effective_date() <= datetime.date.today() and obj.get_expiration_date() > datetime.date.today():
    print('Something is still active')
```

I find it much simpler this way:
``` python
if obj.get_effective_date() <= datetime.date.today() < obj.get_expiration_date():
    print('Something is still active')
```

**Comment by sushobhit27** [comment]
>BTW: The "and" form of the code above has a race condition, because you are calling today() twice and it might give different values, something you got rid of that way too.

``` python
obj = Something()
today = datetime.date.today()
if obj.get_effective_date() <= today() and obj.get_expiration_date() > today():
    print('Something is still active')
```

and yes its a matter of taste and getting used to different syntax.
